cdragontoolbox/sknfile.py

```diff
@@ -0,0 +1,83 @@
+from .tools import BinParser
+
+
+class SknFile:
+    def __init__(self, file):
+        if isinstance(file, str):
+            file = open(file, "rb")
+        if file.read(4) != b"\x33\x22\x11\x00":
+            raise ValueError("missing magic code")
+
+        f = BinParser(file)
+
+        self.major, self.minor = f.unpack("<HH")
+
+        if self.major == 0:
+            self.index_count, self.vertex_count = f.unpack("<II")
+            indices = [f.unpack("<H")[0] for i in range(self.index_count)]
+            vertices = [self.read_vertex(f) for i in range(self.vertex_count)]
+            self.entries = [{"name": "Unknown", "vertices": vertices, "indices": [(x + 1) for x in indices]}]
+            return
+
+        count, = f.unpack("<I")
+        self.entries = [self.read_object(f) for i in range(count)]
+
+        if self.major == 4:
+            self.unknown, = f.unpack("<I")
+
+        self.index_count, self.vertex_count = f.unpack("<II")
+
+        if self.major == 4:
+            self.vertex_size, = f.unpack("<I")
+            self.contains_tangent = bool(f.unpack("<I")[0])
+            self.bounding_box_min = f.unpack("<fff")
+            self.bounding_box_max = f.unpack("<fff")
+            self.bounding_sphere_location = f.unpack("<fff")
+            self.bounding_sphere_radius, = f.unpack("<f")
+
+        indices = [f.unpack("<H")[0] for i in range(self.index_count)]
+        vertices = [self.read_vertex(f) for i in range(self.vertex_count)]
+
+        for entry in self.entries:
+            entry["vertices"] = vertices[entry["start_vertex"] : entry["start_vertex"] + entry["vertex_count"]]
+            entry["indices"] = [
+                (x + 1) - (0 if x < entry["start_vertex"] else entry["start_vertex"]) for x in indices[entry["start_index"] : entry["start_index"] + entry["index_count"]]
+            ]
+            
+            # remove redundant information
+            entry.pop("start_vertex", None)
+            entry.pop("start_index", None)
+            entry.pop("vertex_count", None)
+            entry.pop("index_count", None)
```

Comment:
You can simply use `del entry["start_vertex"]`. `pop()` is meant to remove and return the value. Here you don't need it.